### ISSUE TITLE
Can't delete a dataset draft

### ISSUE THREAD

**Issue by wfenrich** [CeON/fairchive#1315]

**Comment by wfenrich** [comment]
Neither as a Superuser, nor as a Deponent. An unknow error is thrown. 

After that I'm also unable to delete files from this draft.

**Comment by madryk** [comment]
Issue is related to fix in task: #1308 
```
Caused by: javax.persistence.OptimisticLockException: Exception [EclipseLink-5010] (Eclipse Persistence Services - 2.6.1.v20150605-31e8258): org.eclipse.persistence.exceptions.OptimisticLockException
Exception Description: The object [edu.harvard.iq.dvn.core.study.FileMetadata[id=3400]] cannot be merged because it has changed or been deleted since it was last read. 
Class> edu.harvard.iq.dataverse.persistence.datafile.FileMetadata
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.mergeInternal(EntityManagerImpl.java:555)
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.merge(EntityManagerImpl.java:530)
	at com.sun.enterprise.container.common.impl.EntityManagerWrapper.merge(EntityManagerWrapper.java:305)
	at edu.harvard.iq.dataverse.engine.command.impl.DeleteDataFileCommand.executeImpl(DeleteDataFileCommand.java:220)
	at edu.harvard.iq.dataverse.engine.command.AbstractVoidCommand.execute(AbstractVoidCommand.java:29)
	at edu.harvard.iq.dataverse.engine.command.AbstractVoidCommand.execute(AbstractVoidCommand.java:1)
	at edu.harvard.iq.dataverse.EjbDataverseEngine.submit(EjbDataverseEngine.java:251)
	at edu.harvard.iq.dataverse.EjbDataverseEngine$1$1.submit(EjbDataverseEngine.java:440)
	at edu.harvard.iq.dataverse.engine.command.impl.DestroyDatasetCommand.executeImpl(DestroyDatasetCommand.java:76)
	at edu.harvard.iq.dataverse.engine.command.AbstractVoidCommand.execute(AbstractVoidCommand.java:29)
	at edu.harvard.iq.dataverse.engine.command.AbstractVoidCommand.execute(AbstractVoidCommand.java:1)
	at edu.harvard.iq.dataverse.EjbDataverseEngine.submit(EjbDataverseEngine.java:251)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.ejb.security.application.EJBSecurityManager.runMethod(EJBSecurityManager.java:1081)
	at org.glassfish.ejb.security.application.EJBSecurityManager.invoke(EJBSecurityManager.java:1153)
	at com.sun.ejb.containers.BaseContainer.invokeBeanMethod(BaseContainer.java:4786)
	at com.sun.ejb.EjbInvocation.invokeBeanMethod(EjbInvocation.java:656)
```
